EIPS/eip-4750.md

```diff
@@ -103,18 +94,16 @@ If the code is valid EOF1, the following execution rules apply:
 #### `JUMPF`
 
 1. Has one immediate argument,`code_section_index`, encoded as a 16-bit unsigned big-endian value.
-2. If data stack has less than `caller_stack_height + type[code_section_index].inputs` items, execution results in exceptional halt.
-3. If data stack has more than `caller_stack_height + type[code_section_index].inputs` items, discards the items between `caller_stack_height` and top `type[code_section_index].inputs` items, so that there are exactly `caller_stack_height +  type[code_section_index].inputs` items left.
-4. Charges 4 gas.
+2. If data stack does not equal `caller_stack_height + type[code_section_index].inputs` items, execution results in exceptional halt.
```

Comment:
```suggestion
2. If data stack does not equal `caller_stack_height - type[code_section_index].inputs + type[code_section_index].outputs` items, execution results in exceptional halt.
```

Reply:
JUMPF

Reply:
In this case condition 2 would be redundant.

Reply:
I think I understand now that both are needed, but it looks very confusing because equality requirement is always more strict than "less than" requirement... But here the order matters.

I think this combination would be equivalent:
Your number 3
```
3. If data stack does not equal `caller_stack_height + type[current_section_index].outputs + type[code_section_index].inputs - type[code_section_index].outputs` items, execution results in exceptional halt.
```
plus additional requirement for JUMPF instruction validation:
```
-  `type[current_section_index].outputs >= type[code_section_index].inputs - type[code_section_index].outputs` 
```
^ this should replace current requirement `type[current_section_index].outputs == type[code_section_index].outputs` (equal number of outputs for JUMPF caller and callee)

Reply:
These make sense, but per our conversation offline, I'm going to remove JUMPF from 4750 so that we can introduce it separately on top of 5450.